### ISSUE TITLE
KeyError: 'class_args' when testing with class-based views

### ISSUE THREAD

**Issue by stevenhair** [alecthomas/flask_injector#14]
When running unittest with class-based views under Python 3.5 with flask_injector 0.6.2, I get the following error:

```
Error
Traceback (most recent call last):
  File "/Users/shair/repos/flask-injection-bug/tests.py", line 14, in setUp
    FlaskInjector(app=app, modules=[configure])
  File "/Users/shair/.virtualenvs/flask-injection-bug/lib/python3.5/site-packages/flask_injector.py", line 221, in __init__
    process_dict(container, injector)
  File "/Users/shair/.virtualenvs/flask-injection-bug/lib/python3.5/site-packages/flask_injector.py", line 244, in process_dict
    d[key] = wrap_fun(value, injector)
  File "/Users/shair/.virtualenvs/flask-injection-bug/lib/python3.5/site-packages/flask_injector.py", line 63, in wrap_fun
    class_args = fun_closure['class_args']
KeyError: 'class_args'
```

Here is code to reproduce the error:

app.py:
```python
from flask import Flask
from flask.views import View
from flask_injector import FlaskInjector
from injector import inject

app = Flask(__name__)

class Message(object):
    def __init__(self, message):
        self.message = message
    def __str__(self):
        return self.message

class IndexView(View):
    @inject(message=Message)
    def __init__(self, message):
        super(IndexView, self).__init__()
        self.message = message

    def dispatch_request(self):
        return str(self.message)


class ConfigureBinder(object):
    def __init__(self, message):
        self.message = message

    def configure_injector(self, binder):
        binder.bind(Message, to=Message(self.message))

app.add_url_rule('/', view_func=IndexView.as_view('index'))

configure = ConfigureBinder('hello').configure_injector
FlaskInjector(app=app, modules=[configure])

if __name__ == '__main__':
    app.run(debug=True)
```

tests.py:
```python
from unittest import TestCase
from flask_injector import FlaskInjector
from app import app, ConfigureBinder

class TestApp(TestCase):

    def setUp(self):
        app.testing = True
        configure = ConfigureBinder('testing').configure_injector
        FlaskInjector(app=app, modules=[configure])

    def test_message(self):
        response = app.test_client().get('/')
        self.assertIn('testing', str(response.data))
```

The problem seems to lie in injecting multiple times. Possibly related to https://github.com/alecthomas/flask_injector/issues/6?

Update: The offending line was indeed introduced in the fix for https://github.com/alecthomas/flask_injector/issues/6: https://github.com/alecthomas/flask_injector/commit/57fddc270020ab4c046b2e559ee8b122d71a7dca#diff-76d5d260b4726f4ecf4f78cba40a6597R50

Update 2: [Here is root offending line](https://github.com/alecthomas/flask_injector/blob/master/flask_injector.py#L115). The problem is that class_args is not passed into `injector.create_object`, so [we don't get a `class_args` property from `fun.__closure__`  in `wrap_class_based_view`](https://github.com/alecthomas/flask_injector/blob/master/flask_injector.py#L63).

**Comment by stevenhair** [comment]
Fix here: https://github.com/alecthomas/flask_injector/pull/15

**Comment by jstasiak** [comment]
I actually think the issue is with the code you present because it seems to be creating multiple ``FlaskInjector`` instances with the same Flask application object which, while not explicitly disallowed, is not a supported use case and I recommend not doing it. I will think about raising an exception if situation like this is detected.

Using class-based views (whether in tests or in a "real" situation) has been supported by this module for a while now (see [1]), on Python 3.5 too.

[1] https://github.com/alecthomas/flask_injector/blob/d194c20ec5658e7bc3d958c3809404bd7cbbad95/flask_injector_tests.py#L37

**Comment by stevenhair** [comment]
No, I completely agree with you. Injecting multiple times doesn't really make sense, which I figured out after spending more time with flask_injector/injector. I was actually going to close https://github.com/alecthomas/flask_injector/pull/15, but @alecthomas merged it in before I got around to closing it.

I agree that some sort of warning or exception would be probably be better and promote better project structure.
